jenkins-integration/beaker/install/shared/r10k_deploy.rb

```diff
@@ -73,9 +80,22 @@ def run_r10k_deploy(host, r10k_config)
 step "install and configure r10k" do
   r10k_config = get_r10k_config_from_env()
   if r10k_config
-    install_r10k(master)
+    install_r10k(master, bin, r10k_version)
     create_r10k_config(master, r10k_config)
-    run_r10k_deploy(master, r10k_config)
+
+    # This is straight up horrific.  Sorry.
+    # Old versions of r10k can do fun things like throw NPEs and return a non-zero
+    # exit code even when they've actually successfully deployed.  If you then
+    # simply re-run the same r10k command afterward, it'll get a zero exit code.
+    # So, we will do an initial deploy and then if we got any non-zero exit codes,
+    # we'll try those environments one more time.  Sorry.
+    results = run_r10k_deploy(master, bin, r10k_config[:environments],
+                              {:acceptable_exit_codes => [0, 1]})
+    failed_envs = results.select { |env,exit_code| exit_code == 1 }.keys
+    if failed_envs.size > 0
+      Beaker::Log.warn("R10K deploy failed on environments: #{failed_envs}; trying one more time.")
+      run_r10k_deploy(master, bin, failed_envs, {})
```

Comment:
If it fails after running a second time, should it cause the beaker script to blow up?